File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*; 
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int left = 0;
        int right = people.length - 1;
        for (int i = 0; i < people.length; i++) {
            if(left > right) break;

            if(people[left] + people[right] <= limit){
                left++;
            }
            right--;
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.rbegin(),people.rend());

    int start=0,end=people.size()-1;
    
    while(start<end){
        if(people[start]+people[end]<=limit)
        {
            end--;
        }
        answer++;
        start++;
    }
    if(start==end)
        answer++; 
    return answer;
}